testsuite/features/step_definitions/navigation_steps.rb

```diff
@@ -1109,3 +1109,21 @@
 Then(/^I should see the text "(.*?)" in the (Operating System|Architecture|Channel Label) field/) do |text, field|
   page.has_field?(text, with: field)
 end
+
+Then(/^I should see the correct timestamp for task "([^"]*)"/) do |task_name|
+  now = Time.now
+  execute_script 'window.stop()'
+  # find row with corresponding task name
+  page.find_all(:xpath, "//table[@class='table table-responsive']//tr").each do |tr|
+    next unless tr.has_text?(task_name)
+    # if task name is found, iterate through the columns to find the timestamp
+    page.find_all(:xpath, "//table[@class='table table-responsive']//td").each do |td|
+      # if a text matching the format xx:xx is found, get and save the text
+      next unless td.text.match(/\d{2}:\d{2}/)
+      if td.text.match(/\d{2}:\d{2}/)
```

Comment:
Couldn't this `if`statement not being removed since you already check the same in the line above? You skip if you do not find the `td.text.match()` so you can be sure to have found it in the next line couldn't you?

Reply:
you're right that if is redundant, thanks!